### PR TITLE
test: Simplify CI checks

### PR DESCRIPTION
**Pull request by giovannicimolin**
This PR simplifies the CI checks and removes CI checks for Python 3.5 (Juniper release and earlier).

**Reviewer:**
- [ ] @xitij2000 (as FF) OR 
- [ ] @alfredchavez (working on a related task in the sprint) OR
- [x] @Agrendalath (since you commented about the CI :stuck_out_tongue:)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -77,29 +32,29 @@ jobs:
...
       - checkout
       - restore_cache:
-          keys: python3.5
+          keys: python3.8
       - requirements
       - run:
-          name: Run the PII checks
-          command: "./venv/bin/tox -e pii_check"
+          name: Run PII checks
+          command: 'tox -e pii_check'
```
Comment (Agrendalath) [inline .circleci/config.yml]: Nit: I believe these quotes are redundant.

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,71 +1,26 @@
...
-          name: Install virtualenv
-          command: sudo apt-get update && sudo apt-get -y -q install virtualenv
-      - run:
-          name: Create a virtualenv environment
-          command: "virtualenv -p python<<parameters.python_version>> venv"
-
       - run:
           name: Install requirements
-          command: ". venv/bin/activate && make requirements"
+          command: "sudo make requirements"
```
Comment (Agrendalath) [inline .circleci/config.yml]: @giovannicimolin, personally, I  like using virtualenvs in the CI, as it allows easier caching [like this](https://gitlab.com/opencraft/dev/sprints/-/blob/master/.gitlab-ci.yml#L44-53).

**Review by Agrendalath — APPROVED** [review]
I've left two optional comments, but they are not blocking.

:+1:
Looks much better!

* [x]  I tested this: checked that the CI passes
* [x]  I read through the code
* [x]  I checked for accessibility issues: n/a
* [x]  Includes documentation: n/a
* [x] I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository: n/a

File: .circleci/config.yml (hunk 1 above)
Comment (giovannicimolin) [reply .circleci/config.yml]: Yes, they were. I've removed them in the latest commit.

File: .circleci/config.yml (hunk 2 above)
Comment (giovannicimolin) [reply .circleci/config.yml]: @Agrendalath Makes sense, I've reverted back to a virtualenv and added the `venv` folder to the caching settings.

**Pull request merged**